# Monte-Carlo Ternary Fission Simulator

This cell imports the necessary packages for the code
as well as a library which contains the code itself

In [1]:
%pylab inline
import numpy as np
import matplotlib.pyplot as plt
from mcternary import *

Populating the interactive namespace from numpy and matplotlib


The following is the main function of the program. It
takes the mass number of the desired nucleus and 
number of events. It also takes the names of the arrays
which contain the variables used throughout the code.
Using these inputs, the function calls all of the helper
functions in the library, and executes them in the correct
sequence. Subsequently, the function returns the initial
and final values of the coordinates, velocities and
momenta of each of the fragments.

In [4]:
runMCTernary(239, 4)

8.92773891679
20.8164698315
24.0574790827
7.92773891679
19.8164698315
23.0574790827
6.92773891679
18.8164698315
22.0574790827
5.92773891679
17.8164698315
21.0574790827
4.92773891679
16.8164698315
20.0574790827
3.92773891679
15.8164698315
19.0574790827
2.92773891679
14.8164698315
18.0574790827
1.92773891679
13.8164698315
17.0574790827
0.927738916793
12.8164698315
16.0574790827
-0.0722610832073
11.8164698315
15.0574790827
Results
8.92773891679
20.8021886035
22.7616901029
7.92773891679
19.8021886035
21.7616901029
6.92773891679
18.8021886035
20.7616901029
5.92773891679
17.8021886035
19.7616901029
4.92773891679
16.8021886035
18.7616901029
3.92773891679
15.8021886035
17.7616901029
2.92773891679
14.8021886035
16.7616901029
1.92773891679
13.8021886035
15.7616901029
0.927738916793
12.8021886035
14.7616901029
-0.0722610832073
11.8021886035
13.7616901029
Results
8.92773891679
20.8691277912
24.1635239467
7.92773891679
19.8691277912
23.1635239467
6.92773891679
18.8691277912
22.1635239467
5.92773891

This program follows the methods presented by Radi et al. in Monte Carlo Studies of Alpha-accompanied
Fission. It does so by taking the mass number, A, of the desired nucleus along with the number of events, 
and going through the following steps:

1. Performing a Monte-Carlo simulation for the mass number and the charge of the light fragment, AL, ZL.
    Values for mu and sigma are obtained from (reference TBD)
2. Using the semi-empirical mass formula to find the mass of the light fragment, mL.
3. Given a fixed A, and the obtained values for AL and ZL, the we get the mass number and charge of the 
    heavy fragment, assuming Aalapha and Zalpha to be 4 and 2, respectively.
4. Using the semi-empirical mass formula to obtain mH. (and alpha?)
5. (energies at scission by MC)
6. Using equation 21 from the paper to find values for d0 and D0 which will later be used to find the separation
    distance between L and the alpha plane, and the distance between L and H, d and D.
7. Running a MC simulation of d and D using d0 and D0 as mu, sigma_D=1 and sigma_d=sigma_xp=0.93.
8. Using equation 29 to find the coordinates of the prime axes, and setting the distance between alpha and the
    x-axis, Rc, equal to y'.
9. Using equaion 30 to find the momentum about the prime axes.
10. Calculating the initial velocities of L and H, VL0 and VH0 from the conservation of momentum.
11. Using equations 11-13 to calculate the constants, R0, V0, and T0. We obtain valuses for RL and RH from
    (reference TBD).
12. Calculating the initial coordinates and velocity components for the three fragments using equations from
    table I.
13. Using equations 7-10 to get the dimentionless variables required for the differential equations.
14. We get the total momenta of the three fragments and the distance between alpha and both, L and H. These
    will be used later to assesss the results of solving the equations of motion.
15. A function is written that calls all of the previous functions. This function initializes all the paramaters
    before they are passed into the numerical solver.
16. Another function is written that solves the differentail equations once the initial parameters are passed to it.
17. We write a third function that writes the results into file.
18. All of these functions are called within the main function which executes them in the correct sequence.
    While assessing the results coming out of the numerical solver. Where if the position of alpha passes
    within 0.8 of the radius of either of the other fragments, the program discards that event, and goes back
    to initialize new parameters. Additionally, the program checks the change in the momentum of the alpha
    particle, and to stop or continue calculating accordingly. Finally, this main function writes into file the
    initial and final values of the coordinates, velocities and momenta of the three fragments.